mobilenet-burn/src/model/mobilenet.rs

```diff
@@ -0,0 +1,169 @@
+use alloc::vec;
+use alloc::vec::Vec;
+use burn::{
+    config::Config,
+    module::Module,
+    nn::{
+        pool::{AdaptiveAvgPool2d, AdaptiveAvgPool2dConfig},
+        Dropout, DropoutConfig, Linear, LinearConfig,
+    },
+    tensor::{backend::Backend, Tensor},
+};
+
+use crate::model::{
+    conv_norm::Conv2dNormActivationConfig, inverted_residual::InvertedResidualConfig,
+    utils::make_divisble,
+};
+
+use super::{
+    conv_norm::{Conv2dNormActivation, NormalizationType},
+    inverted_residual::InvertedResidual,
+};
+
+#[derive(Debug, Module)]
+pub struct MobileNetV2<B: Backend> {
+    features: Vec<FeatureLayersType<B>>,
+    classifier: Vec<ClassifierLayersType<B>>,
+    avg_pool: AdaptiveAvgPool2d,
+}
+
+#[derive(Debug, Config)]
+pub struct MobileNetV2Config {
+    #[config(default = "1000")]
+    num_classes: usize,
+
+    #[config(default = "1.0")]
+    width_mult: f32,
+
+    #[config(default = "vec![vec![]]")]
+    inverted_residual_setting: Vec<Vec<usize>>,
+
+    #[config(default = "8")]
+    round_nearest: usize,
+
+    norm_layer: NormalizationType,
+
+    #[config(default = "0.2")]
+    dropout: f64,
+}
+#[derive(Module, Debug)]
+enum FeatureLayersType<B: Backend> {
+    Block(InvertedResidual<B>),
+    Conv(Conv2dNormActivation<B>),
+}
```

Comment:
Feature layers are convolutional blocks, so we could rename the enum to something `ConvBlock` to be a bit more explicit.

```rust
enum ConvBlock<B: Backend> {
    InvertedResidual(InvertedResidual<B>),
    Conv(Conv2dNormActivation<B>),
}
```